### PR TITLE
Searching: getElement*, querySelector*

### PR DESCRIPTION
**Pull request by shahrzadJavadiKoushesh**

### CODE REVIEW

**Review by alivarastepour — CHANGES_REQUESTED** [review]
please re-format your translation so that the number of lines match the number of lines in the original version.

**Comment by mahdiHash** [comment]
Hi, @shahrzadJavadiKoushesh.
Thanks for your help.
Please translate the tasks of the article, too (the files in [1-find-elements](https://github.com/javascript-tutorial/fa.javascript.info/tree/master/2-ui/1-document/04-searching-elements-dom/1-find-elements) directory).

**Comment by mahdiHash** [comment]
Do you like to review the whole article, @alivarastepour?

**Comment by alivarastepour** [comment]
> Do you like to review the whole article, @alivarastepour?

sure. will do.

**Comment by mahdiHash** [comment]
> > Do you like to review the whole article, @alivarastepour?
> 
> sure. will do.

Thank you. If there's any question, let me know.

**Comment by shahrzadJavadiKoushesh** [comment]
> Hi, @shahrzadJavadiKoushesh. Thanks for your help. Please translate the tasks of the article, too (the files in [1-find-elements](https://github.com/javascript-tutorial/fa.javascript.info/tree/master/2-ui/1-document/04-searching-elements-dom/1-find-elements) directory).

/done

**Review by alivarastepour — CHANGES_REQUESTED** [review]
Hi,
you have forgotten to merge your changes to the current branch; also note that top file metadata should not be translated.

**Comment by shahrzadJavadiKoushesh** [comment]
> Hi, you have forgotten to merge your changes to the current branch; also note that top file metadata should not be translated.

/done

**Review by alivarastepour — CHANGES_REQUESTED** [review]
Hello again,
First of all, thanks for your contribution and time devoted to this project;
As you have probably guessed, using markdown with two languages which have different directions is a bit challenging; so we kindly ask you to follow the steps mentioned in this [repo ](https://github.com/javascript-tutorial/server)to bring a local server up and see your changes as they are made to each individual markdown.
all changes should maintain the readability of document; currently, that is not the case with your changes.
Thanks again

**Comment by mahdiHash** [comment]
> Hello again, First of all, thanks for your contribution and time devoted to this project; As you have probably guessed, using markdown with two languages which have different directions is a bit challenging; so we kindly ask you to follow the steps mentioned in this [repo ](https://github.com/javascript-tutorial/server)to bring a local server up and see your changes as they are made to each individual markdown. all changes should maintain the readability of document; currently, that is not the case with your changes. Thanks again

Checking the translation with the server repo isn't that necessary. You can read the article on github since it'll keep the RTL format of Persian (although in some cases it's overridden, but it's ok overall).
So, please submit your review on this translation, @alivarastepour.

File: 2-ui/1-document/04-searching-elements-dom/article.md [hunk 1]
```diff
@@ -1,84 +1,84 @@
...
-```warn header="Only `document.getElementById`, not `anyElem.getElementById`"
-The method `getElementById` can be called only on `document` object. It looks for the given `id` in the whole document.
+```warn header=" فقط `document.getElementById`, نه `anyElem.getElementById`"
+.داده شده می‌گردد `id` به دنبال document صدا بزنیم. آن در تمام document را روی اشیای `getElementById‍‍` ما تنها می‌توانیم
 ```
 
 ## querySelectorAll [#querySelectorAll]
 
-By far, the most versatile method, `elem.querySelectorAll(css)` returns all elements inside `elem` matching the given CSS selector.
+تا اینجا، همه‌کاره‌ترین روش `elem.querySelectorAll(css)` بود که تمام elementهای درون `elem` که با css selector داده شده تطابق دارند را انتخاب می‌کند.
```
Comment (alivarastepour) [inline 2-ui/1-document/04-searching-elements-dom/article.md:79]:
```suggestion

با اختلاف، جامع‌ترین روش جستجو، `elem.querySelectorAll(css)` است که تمام elementهای درون `elem` که با css selector داده شده تطابق دارند را انتخاب می‌کند. 
```

File: 2-ui/1-document/04-searching-elements-dom/article.md [hunk 2]
```diff
@@ -100,25 +100,25 @@ Here we look for all `<li>` elements that are last children:
...
-Previous methods were searching the DOM.
+روش‌های قبلی در DOM جستجو می‌کردند. 
 
-The [elem.matches(css)](https://dom.spec.whatwg.org/#dom-element-matches) does not look for anything, it merely checks if `elem` matches the given CSS-selector. It returns `true` or `false`.
+عبارت [elem.matches(css)](https://dom.spec.whatwg.org/#dom-element-matches) در جستجوی چیزی نیست، فقط چک می‌کند که element با CSS-selector داده شده مطابقت دارد یا نه و `true` یا `false` برمی‌گرداند. 
 
-The method comes in handy when we are iterating over elements (like in an array or something) and trying to filter out those that interest us.
+این روش زمانی به کار می‌آید که می‌خواهیم روی elementهایی (مثل آرایه یا همچین چیزی) iterate کنیم و آن‌هایی که مطلوب هستند را فیلتر کنیم. 
 
 For instance:
```
Comment (alivarastepour) [inline 2-ui/1-document/04-searching-elements-dom/article.md:123]:
```suggestion
برای مثال:
```

File: 2-ui/1-document/04-searching-elements-dom/article.md [hunk 3]
```diff
@@ -164,44 +164,46 @@ For instance:
...
 </script>
 ```
 
 ## getElementsBy*
 
-There are also other methods to look for nodes by a tag, class, etc.
+همچنین روش‌های دیگری برای جستجو برای یک node با tag، class و ... وجود دارد. 
 
-Today, they are mostly history, as `querySelector` is more powerful and shorter to write.
+امروزه آن‌ها بیشتر تاریخچه هستند چون `querySelector` قدرتمندتر است و برای نوشتن سریع‌تر است.
```
Comment (alivarastepour) [inline 2-ui/1-document/04-searching-elements-dom/article.md:175]:
```suggestion
امروزه آن‌ها کمتر استفاده می‌شوند چون `querySelector` قدرتمندتر است و برای نوشتن سریع‌تر است. 
```

File: 2-ui/1-document/04-searching-elements-dom/article.md [hunk 4]
```diff
@@ -363,12 +365,12 @@ There are 6 main methods to search for nodes in DOM:
 </tbody>
 </table>
 
-By far the most used are `querySelector` and `querySelectorAll`, but `getElement(s)By*` can be sporadically helpful or found in the old scripts.
+.های قدیمی یافت شود script هم گاهی به صورت پراکنده می‌تواند مفید باشد یا در `getElement(s)By` شده است، اما `querySelectorAll` و `querySelector` تا اینجا بیشترین استفاده از
```
Comment (alivarastepour) [inline 2-ui/1-document/04-searching-elements-dom/article.md:368]:
```suggestion
.های قدیمی یافت شود script هم گاهی به صورت پراکنده می‌تواند مفید باشد یا در `getElement(s)By` شده است، اما `querySelectorAll` و `querySelector` با اختلاف بیشترین استفاده از
```

File: 2-ui/1-document/04-searching-elements-dom/article.md [hunk 5]
```diff
@@ -363,12 +365,12 @@ There are 6 main methods to search for nodes in DOM:
...
 
-- There is `elem.matches(css)` to check if `elem` matches the given CSS selector.
-- There is `elem.closest(css)` to look for the nearest ancestor that matches the given CSS-selector. The `elem` itself is also checked.
+.تطابق دارد یا نه CSS-selector با elem برای این استفاده می‌شود که چک کنیم `elem.matches(css)` از عبارت -
+.هم چک می‌شود `elem` داده شده تطابق دارند. خود CSS-selector برای جستجوی نزدیک‌ترین والدهایی است که با `elem.closest(css)` عبارت - 
 
-And let's mention one more method here to check for the child-parent relationship, as it's sometimes useful:
--  `elemA.contains(elemB)` returns true if `elemB` is inside `elemA` (a descendant of `elemA`) or when `elemA==elemB`.
+و بیایید اینجا یک روش دیگر برای بررسی رابطه والد و فرزند را چک کنیم،‌چون بعضی وقت‌ها مفید است: 
+.را برمی‌گرداند true مثدار `elemA.contains(elemB)` باشد، عبارت `elemA==elemB` یا وقتی که (`elemA` از فرزندان) باشد `elemA` داخل `elemB` اگر -
```
Comment (alivarastepour) [inline 2-ui/1-document/04-searching-elements-dom/article.md:376]:
```suggestion
.را برمی‌گرداند true مقدار `elemA.contains(elemB)` باشد، عبارت `elemA==elemB` یا وقتی که (`elemA` از فرزندان) باشد `elemA` داخل `elemB` اگر -
```

File: 2-ui/1-document/04-searching-elements-dom/article.md [hunk 6]
```diff
@@ -1,84 +1,84 @@
...
-
+  let elem = 5; // <div id="elem"> پنج است، نه ارجاعی به elem الان مقدار
+ 
   alert(elem); // 5
 </script>
 ```
 
-```warn header="Please don't use id-named global variables to access elements"
-This behavior is described [in the specification](https://html.spec.whatwg.org/multipage/window-object.html#named-access-on-the-window-object), but it is supported mainly for compatibility.
+```warn header=".ها استفاده نکنیدelement برای دسترسی به global لفطا از متغیرهای"
```
Comment (alivarastepour) [inline 2-ui/1-document/04-searching-elements-dom/article.md:57]:
```suggestion
```warn header=".ها استفاده نکنیدelement برای دسترسی به global لطفا از متغیرهای"
```

File: 2-ui/1-document/04-searching-elements-dom/1-find-elements/task.md [hunk 7]
```diff
@@ -2,17 +2,17 @@ importance: 4
...
 ---
 
-# Search for elements
+# جستجوی elementها
 
-Here's the document with the table and form.
+اینجا document مربوط به table و form را می‌بینیم. 
 
-How to find?...
+چگونه پیدا کنیم؟
```
Comment (alivarastepour) [inline 2-ui/1-document/04-searching-elements-dom/1-find-elements/task.md:9]:
```suggestion
چگونه ... را پیدا کنیم؟
```

**Review by alivarastepour — CHANGES_REQUESTED** [review]
Hi @shahrzadJavadiKoushesh , i have proposed some changes to improve your translation even further. please review them.

File: 2-ui/1-document/04-searching-elements-dom/article.md (hunk 1 above)
Comment (mahdiHash) [reply 2-ui/1-document/04-searching-elements-dom/article.md:79]: > 

```suggestion
تا اینجا، جامع‌ترین روش جستجو `elem.querySelectorAll(css)` بود که تمام elementهای درون `elem` که با css selector داده شده تطابق دارند را انتخاب می‌کند. 
```
"By far" translation is correct.

File: 2-ui/1-document/04-searching-elements-dom/article.md (hunk 4 above)
Comment (mahdiHash) [reply 2-ui/1-document/04-searching-elements-dom/article.md:368]: > 

The translation is ok, but aren't the words disordered?
It's not a code comment, so if you've changed the order of translation, please turn it back.

File: 2-ui/1-document/04-searching-elements-dom/1-find-elements/task.md (hunk 7 above)
Comment (mahdiHash) [reply 2-ui/1-document/04-searching-elements-dom/1-find-elements/task.md:9]: > 

```suggestion
چگونه موارد زیر را پیدا کنیم؟
```
The [ellipsis](https://en.wikipedia.org/wiki/Ellipsis) refers to the follow-up lines.

**Comment by mahdiHash** [comment]
Thanks for the review, @alivarastepour :D

**Review by mahdyar — APPROVED** [review]
LGTM!

**Review by alivarastepour — APPROVED** [review]

**Comment by mahdiHash** [comment]
Thanks for your contribution, @shahrzadJavadiKoushesh :D

**Review by mahdiHash — APPROVED** [review]

**Pull request merged**